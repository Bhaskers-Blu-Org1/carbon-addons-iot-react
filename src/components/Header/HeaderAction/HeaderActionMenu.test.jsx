import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import HeaderActionMenu from './HeaderActionMenu';

describe('HeaderActionMenu', () => {
  const mockProps = {
    'aria-label': 'Accessibility label',
    className: 'custom-class',
    // We use `ref` instead of `focusRef` becase `HeaderActionMenu` forwards the ref
    // to the underlying menu button
    ref: jest.fn(),
    tabIndex: -1,
    onToggleExpansion: jest.fn(),
    label: 'my-menu',
    childContent: [
      {
        metaData: {
          href: 'http://google.com',
          title: 'this is a title',
          target: '_blank',
          rel: 'noopener noreferrer',
          element: 'a',
        },
        content: 'this is my message to you',
      },
      {
        metaData: {
          onClick: jest.fn(),
          className: 'this',
          element: 'button',
        },
        content: <span>JohnDoe@ibm.com</span>,
      },
    ],
  };

  it('should render', () => {
    const { container } = render(<HeaderActionMenu {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render aria-label', () => {
    const { getByText } = render(<HeaderActionMenu {...mockProps} />);

    expect(getByText('Accessibility label')).toBeInTheDocument();
  });

  it('should render content prop', () => {
    const menuContent = () => <p>Some other text</p>;
    const { getByText, queryByText } = render(
      <HeaderActionMenu renderMenuContent={menuContent} {...mockProps} />
    );

    expect(queryByText('Accessibility label')).not.toBeInTheDocument();
    expect(getByText('Some other text')).toBeInTheDocument();
  });
});
