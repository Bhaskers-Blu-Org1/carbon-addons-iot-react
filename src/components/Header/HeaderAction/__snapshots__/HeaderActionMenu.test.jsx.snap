// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderActionMenu should render 1`] = `
<div>
  <div
    class="bx--header__submenu custom-class"
  >
    <a
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="Accessibility label"
      class="bx--header__menu-item bx--header__menu-title"
      data-testid="menuitem"
      href=""
      role="menuitem"
      tabindex="0"
    >
      Accessibility label
      <svg
        aria-hidden="true"
        class="bx--header__menu-arrow"
        focusable="false"
        height="6"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 10 6"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 6L0 1 0.7 0.3 5 4.6 9.3 0.3 10 1z"
        />
      </svg>
    </a>
    <ul
      aria-label="Accessibility label"
      class="bx--header__menu"
      role="menu"
    >
      <li>
        <a
          class="bx--header__menu-item"
          href="http://google.com"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
          title="this is a title"
        >
          <span
            class="bx--text-truncate--end"
          >
            this is my message to you
          </span>
        </a>
      </li>
      <li
        class="this"
      >
        <button
          class="bx--header__menu-item"
          tabindex="0"
        >
          <span
            class="bx--text-truncate--end"
          >
            <span>
              JohnDoe@ibm.com
            </span>
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
